tests/tools/stepper.c

```diff
@@ -142,7 +162,16 @@ int main( int argc, char **argv)
 
       if (debug)
          fprintf( out, "%d [%s]\n", id, query);
-      rc = clnt_run_query(clnt, query, out);
+
+      int retries = 0;
+      while (retries < 10) {
+         rc = clnt_run_query(clnt, query, out);
```

Comment:
I figured out that this was failing on my platform because of `SIGPIPE` signals! After disconnect, the next socket access throws a `SIGPIPE` signal. This exits the program by default. 

You can resolve this by setting the `SIGPIPE` signal handler to `SIG_IGN` like this: `signal(SIGPIPE, SIG_IGN);`. This causes the program to ignore `SIGPIPE` signals. It would be reasonable to do this at the top of `main()` as it is done [here](https://github.com/morgando/comdb2/blob/main/tests/tools/hatest.c#L761).